roles/deprovision-ios-app/tasks/main.yml

```diff
@@ -1,3 +1,11 @@
 # Delete iOS app representation using the mobile cli
+- set_fact: app_not_found_err="'mobileclients.mobile.k8s.io \"{{ appName }}-ios\" not found'"
+
+- name: "Check if {{ appName }} client exists"
+  shell: mobile get client {{ appName }}-ios --namespace={{ namespace }} -o=json
+  register: mobile_client
+  failed_when: mobile_client.stdout == "" and not mobile_client.stderr | search({{ app_not_found_err }})
```

Comment:
use not () e.g. not (mobile_client.stderr | search(app_not_found_err) and you dont need to use {[...}} inside failed_when statement